index.bs

```diff
@@ -168,11 +172,14 @@ In all cases, if present, the length is coded using LEB128 as specified in [[!AV
 
 ISSUE: This field could disappear if we decide to mandate only one mode.
 
-The <dfn>initial_presentation_delay</dfn> field indicates the number of samples that should be decoded prior to starting the presentation of the first sample associated with this sample entry in order to guarantee that each sample will be decoded prior to its presentation time (as indicated by the timing information in ISOBMFF structures) if decoding happens at the decoding rate required by the level indicated in the [=Sequence Header=] and according to the hypothetical decoder model specified in [[!AV1]]. The value of this field shall be compatible with the value of decoding_delay indicated in the [=Sequence Header=].
-
-ISSUE: The initial_presentation_delay might be removed if added to the Sequence Header.
+The <dfn>initial_presentation_delay_minus_one</dfn> field indicates the number of samples (minus one) that should be decoded prior to starting the presentation of the first sample associated with this sample entry in order to guarantee that each sample will be decoded prior to its presentation time under the constraints of the first level value indicated in the [=Sequence Header=] (in the configOBUs field or in the associated samples). More precisely, the following procedure SHALL not return any error:
+- construct a hypothetical bitstream consisting of the OBUs carried in the sample entry followed by the OBUs carried in all the samples,
+- set the first [=initial_display_delay_minus1=] field of each sequence header to the number of frames contained in the first [=initial_presentation_delay_minus_one=] + 1 samples,
+- set the [=frame_presentation_delay=] field of the frame header of each presentable frame such that it matches the presentation time difference between the sample carrying this frame and the previous sample (if it exists, 0 otherwise)
+- apply the display model verification algorithm.
+In particular, when smooth presentation can be guaranteed after decoding the first sample, the value 0 SHALL be used; and if an ISOBMFF writer cannot verify the above procedure, it should set [=initial_presentation_delay=] to its maximum value.
```

Comment:
This looks okay, but are you sure you want to enforce it here? You could alternately just trust the values inside the bitstream itself and specify how they are copied into the container.

Reply:
In my view, bitstreams should not contain any timing information because it might conflict with the ISOBMFF one, after editing for example or slicing/splicing. The patch actually contains text recommending not to set timing_info_present_flag and others similar fields for streams stored in ISOBMFF. Because the field initial_display_delay_minus1 is behind those flags, you cannot rely on the values inside the bitstream. 

However, I think there is value if a muxer can properly set it even if it is not in the bitstream, so I wanted to have a clear conformance check so that muxers do not put any random value. I also provided guidance if they don't know what to put. Does this make sense?

Reply:
Yes, that makes a lot of sense. I also agree that it really shouldn't be present once muxed, so this definition seems fine. This does leave it open for e.g. a annexb-to-mp4 remuxer to copy values and then strip timing as long as it trusts the input too, I suppose. (ideally people just write straight mp4 with no remuxing step, of course)